src/sp/it/pl/util/system/Environment.kt

```diff
@@ -340,5 +339,5 @@ private fun File.openWindowsExplorerAndSelect() {
 private fun File.isExecutable(): Boolean =
         when (Os.current) {
             Os.WINDOWS -> path.endsWith(".exe", true) || path.endsWith(".bat", true)
-            else -> false
+            else -> path.endsWith(".sh")
```

Comment:
Is sh executable on windows? if not please provide proper LINUX clause.

Also, it is simply incorrect that the else branch has logic, it should be else -> false, because else should denote case where no predicate tested true. I.e., try a, b, c, d else -> fail. Else return d is unnecessary shortcircuiting. We want to express the logic of the alrogithm, not the final logical function. I.e. the simplest boolean function is not always the most useful. Readability is most important.

Reply:
Ah I didnt realize the when executes on Os.current. The makes solution correct.

Reply:
So this is fine with you? Because .sh is executable on any OS except Windows, so it does make sense.

Reply:
Yes. Appears correct.